### PR TITLE
fixed x86-16bit seg:off disassembly print for seg=0

### PR DESCRIPTION
**Pull request by lasek0**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

fixed x86-16bit seg:off disassembly print for seg=0. now analysed function name can be correctly printet in disassembly.
tests done and requested in https://github.com/radareorg/radare2/pull/19607 PR

no issue was opened for that bug.

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -5349,9 +5349,22 @@ static char *ds_sub_jumps(RDisasmState *ds, char *str) {
...
+				ut64 seg = r_num_get (NULL, ptr);
+				ut64 off = r_num_get (NULL, colon + 1);
+				*colon = ':';
+				numval = (seg << seggrn) + off;
+			} else {
+				numval = r_num_get (NULL, ptr);
+			}
 			if (numval == addr) {
-				while (*nptr && !IS_SEPARATOR (*nptr) && *nptr != 0x1b) {
+				while ((*nptr && !IS_SEPARATOR (*nptr) && *nptr != 0x1b) || (x86 && bits == 16 && colon && *nptr == ':')) {
```
Comment (trufae) [inline libr/core/disasm.c:5367]: as long as the conditional is starting to be hrad to read, better to decompose it into another form. try something like:

```c
while (*nptr) {
   // condition to break 
   // the new condition to break
}
```

**Review by trufae — CHANGES_REQUESTED** [review]
lgtm! can you refactor this while condition and add another test for this case? looks good so far! thank you!

**Comment by lasek0** [comment]
Tests for that cases are added in that PR 
https://github.com/radareorg/radare2/pull/19607 In general I added more test cases for previous x86-16 PRs, then I found spare time to try fix one of failing test. I agree that this while condition is hard to read. I will refactor this when I found spare free times.

**Comment by trufae** [comment]
i can merge and improve the loop later if it's ok for you. the code looks good

**Comment by Lazula** [comment]
wrong PR reference, the one i wanted was merged by hand

**Pull request merged**